src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py

```diff
@@ -541,22 +542,24 @@ def set_extension(
                                   protected_settings=protected_settings,
                                   type_handler_version=version,
                                   settings=settings,
-                                  auto_upgrade_minor_version=auto_upgrade_minor_version)
+                                  auto_upgrade_minor_version=(not no_auto_upgrade))
     return client.virtual_machine_extensions.create_or_update(
         resource_group_name, vm_name, vm_extension_name, ext)
 
 def set_vmss_extension(
         resource_group_name, vmss_name, extension_name, publisher,
         version=None, settings=None,
-        protected_settings=None, auto_upgrade_minor_version=False):
+        protected_settings=None, no_auto_upgrade=False):
     '''create/update extensions for a VMSS in a resource group. You can use
     'extension image list' to get extension details
     :param vm_extension_name: the name of the extension
     :param publisher: the name of extension publisher
     :param version: the version of extension.
     :param settings: public settings or a file path with such contents
     :param protected_settings: protected settings or a file path with such contents
-    :param auto_upgrade_minor_version: auto upgrade to the newer version if available
+    :param no_auto_upgrade: by doing this, extension system will not pick the highest minor version
+    for the specified version number, and will not auto update to the latest build/revision number
+    on any scale set updates in future.
```

Comment:
Let us keep the wording, as it has been reviewed by VM linux team; also, it is not a default extension system behavior, rather just our command's default value. FYI, the behavior associated is complex that once you turn on the flag, and later when you update anything on VM, the extension will be updated as well.